Content.Client/_Impstation/CosmicCult/UI/Monument/MonumentMenu.xaml.cs

```diff
@@ -0,0 +1,147 @@
+        EntropyUntilNextStage.Text = Loc.GetString("monument-interface-entropy-value", ("infused", state.EntropyUntilNextStage.ToString()));
+        CrewToConvertUntilNextStage.Text = state.CrewToConvertUntilNextStage.ToString();
+    }
+
+    // Update all the glyph buttons
+    private void UpdateGlyphs()
+    {
+        var glyphs = _glyphPrototypes.ToList();
+        glyphs.Sort((x, y) =>
+            string.Compare(x.Name, y.Name, StringComparison.CurrentCultureIgnoreCase));
+
+        GlyphContainer.RemoveAllChildren();
+        foreach (var glyph in glyphs)
+        {
+            var boxContainer = new BoxContainer();
+            var unlocked = _unlockedGlyphProtoIds.Contains(glyph.ID);
+            var button = new Button
+            {
+                HorizontalExpand = true,
+                StyleClasses = { StyleBase.ButtonSquare },
+                ToolTip = Loc.GetString(glyph.Name),
+                Group = _glyphButtonGroup,
+                Pressed = glyph.ID == _selectedGlyphProtoId,
+                Disabled = !unlocked,
+                Modulate = !unlocked ? Color.Gray : Color.White,
+            };
+            button.OnPressed +=  _ => _selectedGlyphProtoId = glyph.ID;
+            var glyphIcon = new TextureRect
+            {
+                Texture = _sprite.Frame0(glyph.Icon),
+                TextureScale = new Vector2(2f, 2f),
+                Stretch = TextureRect.StretchMode.KeepCentered,
+            };
+            boxContainer.AddChild(button);
+            button.AddChild(glyphIcon);
+            GlyphContainer.AddChild(boxContainer);
+        }
+    }
+
+    // Update all the influence thingies
+    private void UpdateInfluences()
+    {
+        var influences = _influencePrototypes.ToList();
+        influences.Sort((x, y) =>
+            string.Compare(x.Name, y.Name, StringComparison.CurrentCultureIgnoreCase));
+
+        InfluencesContainer.RemoveAllChildren();
+        foreach (var influence in influences)
+        {
+            var unlocked = _unlockedInfluenceProtoIds.Contains(influence.ID);
+            var influenceBox = new InfluenceUIBox(influence, unlocked);
+            influenceBox.OnGainButtonPressed += () => OnGainButtonPressed?.Invoke(influence.ID);
+            if (unlocked == true)
+                InfluencesContainer.AddChild(influenceBox);
+        }
+        foreach (var influence in influences) // I'm sure there's a better way to order them by being Unlocked / Locked, but i've been told to just loop through 'em twice. So we're going with that
+        {
+            var unlocked = _unlockedInfluenceProtoIds.Contains(influence.ID);
+            var influenceBox = new InfluenceUIBox(influence, unlocked);
+            influenceBox.OnGainButtonPressed += () => OnGainButtonPressed?.Invoke(influence.ID);
+            if (unlocked == false)
+                InfluencesContainer.AddChild(influenceBox);
+        }
+    }
```

Comment:
Avoid iterating twice. Now with less it-is-almost-midnight mistakes
```suggestion
    // Update all the influence thingies
    private void UpdateInfluences()
    {
        var influences = _influencePrototypes
            .OrderByDescending(influence => _unlockedInfluenceProtoIds.Contains(influence.ID))
            .ThenBy<InfluencePrototype, string>(influence => influence.Name, StringComparer.CurrentCultureIgnoreCase);

        InfluencesContainer.RemoveAllChildren();
        foreach (var influence in influences)
        {
            var unlocked = _unlockedInfluenceProtoIds.Contains(influence.ID);
            var influenceBox = new InfluenceUIBox(influence, unlocked);
            influenceBox.OnGainButtonPressed += () => OnGainButtonPressed?.Invoke(influence.ID);
            InfluencesContainer.AddChild(influenceBox);
        }
    }
```